completions_test.go

```diff
@@ -3517,3 +3518,78 @@ func TestGetFlagCompletion(t *testing.T) {
 		})
 	}
 }
+
+func TestGetEnvConfig(t *testing.T) {
+	testCases := []struct {
+		use       string
+		suffix    string
+		cmdVar    string
+		globalVar string
+		cmdVal    string
+		globalVal string
+		expected  string
+	}{
+		{
+			use:       "root",
+			suffix:    "test",
+			cmdVar:    "ROOT_TEST",
+			globalVar: "COBRA_TEST",
+			cmdVal:    "cmd",
+			globalVal: "global",
+			expected:  "cmd",
+		},
+		{
+			use:       "root",
+			suffix:    "test",
+			cmdVar:    "ROOT_TEST",
+			globalVar: "COBRA_TEST",
+			cmdVal:    "",
+			globalVal: "global",
+			expected:  "global",
+		},
+		{
+			use:       "root",
+			suffix:    "test",
+			cmdVar:    "ROOT_TEST",
+			globalVar: "COBRA_TEST",
+			cmdVal:    "",
+			globalVal: "",
+			expected:  "",
+		},
+		{
+			use:       "foo.bar",
+			suffix:    "test",
+			cmdVar:    "FOO_BAR_TEST",
+			globalVar: "COBRA_TEST",
+			cmdVal:    "cmd",
+			globalVal: "global",
+			expected:  "cmd",
+		},
+		{
+			use:       "quux-BAZ",
+			suffix:    "test",
+			cmdVar:    "QUUX_BAZ_TEST",
+			globalVar: "COBRA_TEST",
+			cmdVal:    "cmd",
+			globalVal: "global",
+			expected:  "cmd",
+		},
+	}
+
+	for _, tc := range testCases {
+		// Could make env handling cleaner with t.Setenv with Go >= 1.17
+		func() {
```

Comment:
To be able to differentiate between the different tests upon a failure, you can replace `func()` with `t.Run(tc.name, func(t *testing.T)` and add a field `name` to your test struct.
Then you get (because I used `t.use` as the test case name because I was lazy):
```
=== RUN   TestGetEnvConfig/root
    completions_test.go:3591: expected: "cmd", got: ""
=== RUN   TestGetEnvConfig/root#01
    completions_test.go:3591: expected: "global", got: ""
=== RUN   TestGetEnvConfig/root#02
=== RUN   TestGetEnvConfig/foo.bar
    completions_test.go:3591: expected: "cmd", got: ""
=== RUN   TestGetEnvConfig/quux-BAZ
    completions_test.go:3591: expected: "cmd", got: ""
```

Reply:
977823cf93624546e86250c048a94c3f1a4a1f41